--- a/aic8800_fdrv/rwnx_main.c
+++ b/aic8800_fdrv/rwnx_main.c
@@ -5814,6 +5814,9 @@
                                     #if (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 15, 0))
                                         , u32 cac_time_ms
                                     #endif
+                                    #if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 12, 0))
+                                        , int link_id
+                                    #endif
                                         )
 {
     struct rwnx_hw *rwnx_hw = wiphy_priv(wiphy);
--- a/aic8800_fdrv/rwnx_radar.c
+++ b/aic8800_fdrv/rwnx_radar.c
@@ -1399,7 +1399,11 @@
                     #if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 14, 0)
                        &ctxt->chan_def,
                     #endif
+                    #if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 12, 0)
+                       NL80211_RADAR_CAC_FINISHED, GFP_KERNEL, 0);
+                    #else
                        NL80211_RADAR_CAC_FINISHED, GFP_KERNEL);
+                    #endif
     rwnx_send_apm_stop_cac_req(rwnx_hw, radar->cac_vif);
     rwnx_chanctx_unlink(radar->cac_vif);
 
@@ -1499,7 +1503,11 @@
                         #if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 14, 0)
                            &ctxt->chan_def,
                         #endif
+                        #if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 12, 0)
+                           NL80211_RADAR_CAC_ABORTED, GFP_KERNEL, 0);
+                        #else
                            NL80211_RADAR_CAC_ABORTED, GFP_KERNEL);
+                        #endif
         rwnx_chanctx_unlink(radar->cac_vif);
     }
 
